extensions/data-transfer/portability-data-transfer-microsoft/src/main/java/org/datatransferproject/transfer/microsoft/photos/MicrosoftPhotosImporter.java

```diff
@@ -140,7 +140,9 @@ public ImportResult importItem(
   private String createOneDriveFolder(PhotoAlbum album) throws IOException {
 
     Map<String, Object> rawFolder = new LinkedHashMap<>();
-    rawFolder.put("name", album.getName());
+    // clean up album name for microsoft specifically
+    String albumName = album.getName() == null || album.getName().length() == 0 ? "Untitled" : album.getName();
```

Comment:
Length for the single space string will be 1 right? Can we trim and use Strings.isNullOrEmpty?

Reply:
no, it gets trimmed in PhotoAlbum to remove trailing and leading spaces. changed to isnullorempty